tools/libraries/src/Swap.ts

```diff
@@ -0,0 +1,80 @@
+import { ethers, ContractTransaction } from 'ethers'
+import type { Provider } from '@ethersproject/providers'
+import { chainIds } from '@airswap/constants'
+import { Order } from '@airswap/types'
+import { Swap as SwapContract } from '@airswap/swap/typechain/contracts'
+import { Swap__factory } from '@airswap/swap/typechain/factories/contracts'
+import { checkResultToErrors } from '@airswap/utils'
+
+import * as swapDeploys from '@airswap/swap/deploys.js'
+
+export class Swap {
+  public chainId: number
+  public contract: SwapContract
+
+  public constructor(
+    chainId = chainIds.MAINNET,
+    signerOrProvider?: ethers.Signer | Provider
+  ) {
+    this.chainId = chainId
+    this.contract = Swap__factory.connect(
+      Swap.getAddress(chainId),
+      signerOrProvider
+    )
+  }
+
+  public static getAddress(chainId = chainIds.MAINNET): string {
+    if (chainId in swapDeploys) {
+      return swapDeploys[chainId]
+    }
+    throw new Error(`Swap contract not found for chainId ${chainId}`)
+  }
+
+  public async check(
+    order: Order,
+    senderWallet: string,
+    signer?: ethers.providers.JsonRpcSigner
```

Comment:
```suggestion
    signer: ethers.providers.JsonRpcSigner
```

Why make it optional?

Reply:
If a signer is passed to the constructor it'll be used, otherwise one must be passed here.